src/master.rs

```diff
@@ -42,18 +51,28 @@ impl Master {
         };
         ioctl!(master, ec::ioctl::MODULE, &mut module_info)?;
         if module_info.ioctl_version_magic != ec::EC_IOCTL_VERSION_MAGIC {
-            Err(Error::new(
+            return Err(Error::new(
                 ErrorKind::Other,
                 format!(
                     "module version mismatch: expected {}, found {}",
                     ec::EC_IOCTL_VERSION_MAGIC,
                     module_info.ioctl_version_magic
                 ),
-            ))
-        } else {
-            ioctl!(master, ec::ioctl::REQUEST)?;
-            Ok(master)
+            ));
         }
+        Ok(master)
+    }
+
+    pub fn master_count(index: MasterIndex) -> Result<usize> {
```

Comment:
I think you can just pass `0` as the index, since the first master should always exist.